### PR TITLE
feat(sdk): `std.Set` tests

### LINKED ISSUE #2785: [Tracking Issue] Tests for `std` APIs
The PR https://github.com/winglang/wing/pull/2783 introduces placeholders for tests for all `std` types under `sdk_tests`.

- [x] #2953 
- [x] #2956
- [x] #2962
- [x] #2956
- [x] json std tests
- [x] #4363
- [x] number std tests
- [x] #4229
- [x] #4372
- [x] string std tests

### LINKED ISSUE #4372: set std tests

### PR DESCRIPTION
**Pull request by WeepingClown13**
Adding tests for the standard library container type Set.
closes #4372, which in turn also finishes and closes #2785.

## Checklist

- [ ] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [ ] Description explains motivation and solution
- [ ] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

File: examples/tests/invalid/container_types.test.w [hunk 1]
```diff
@@ -29,4 +29,10 @@ let a: Array<str> = MutArray<str>[];
 
 let mm1: MutMap<num> = { "a" => 1, "b" => 2, "c" => 3 };
 let mm2 = MutMap<num> { "a" => 1, "b" => 2, "c" => 3 };
-let mm3 = mm2.copyMut();
\ No newline at end of file
+let mm3 = mm2.copyMut();
+
+
+let ss1: MutSet<str> = Set{"c"};
```
Comment (eladb) [inline examples/tests/invalid/container_types.test.w:35]: We normally add a comment with the expected error

File: tools/hangar/__snapshots__/invalid.ts.snap [hunk 2]
```diff
@@ -759,6 +759,13 @@ error: Unknown parser error
    |                                               ^^^^ Unknown parser error
 
 
+error: Unknown parser error
```
Comment (eladb) [inline tools/hangar/__snapshots__/invalid.ts.snap:762]: This seems like a bug in the parser... @MarkMcCulloh ?

**Review by eladb — APPROVED** [review]

File: tools/hangar/__snapshots__/invalid.ts.snap (hunk 2 above)
Comment (WeepingClown13) [reply tools/hangar/__snapshots__/invalid.ts.snap:762]: @eladb I am troubleshooting it locally and it's for sure a bug. I'm trying out different combinations and am baffled by how some of them passes but not the others.

File: examples/tests/invalid/container_types.test.w (hunk 1 above)
Comment (WeepingClown13) [reply examples/tests/invalid/container_types.test.w:35]: I'll update it.

File: tools/hangar/__snapshots__/invalid.ts.snap (hunk 2 above)
Comment (MarkMcCulloh) [reply tools/hangar/__snapshots__/invalid.ts.snap:776]: The message is bad but the syntax is indeed invalid, `Set{"c"}` should be `Set<str>{"c"}` (or just `{"c"}`)

**Pull request merged**